server/src/main/java/org/elasticsearch/rest/action/document/RestDeleteAction.java

```diff
@@ -66,6 +67,8 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         deleteRequest.setRefreshPolicy(request.param("refresh"));
         deleteRequest.version(RestActions.parseVersion(request));
         deleteRequest.versionType(VersionType.fromString(request.param("version_type"), deleteRequest.versionType()));
+        deleteRequest.setIfMatch(
+            request.paramAsLong("if_seq_no_match", SequenceNumbers.UNASSIGNED_SEQ_NO), request.paramAsLong("if_primary_term_match", 0));
```

Comment:
can you use the values of the getters (`deleteRequest.ifSeqNoMatch()` and `deleteRequest.ifPrimaryTermMatch()`) as fallbacks. This ensures that the defaults on the REST layer agree with those on the transport layer.
See how it's done for the versionType one line above this.

Reply:
+1